### PR TITLE
Do not display unused taxonomies on Work and Blog listings

### PR DESCRIPTION
**Pull request by engineervix**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1414738166)

### Description of Changes Made

This PR makes changes to the Work and Blog listings so that only Service and Sector taxonomies that are associated with the child pages are available on the listing for filtering.

By doing this, we avoid returning zero results when one tries to apply a filter that is not associated with any Page, as documented on the [ticket](https://torchbox.monday.com/boards/1192293412/pulses/1414738166). 

### How to Test

1. Pull latest staging data if you haven't already done so
2. Add at least 1 service taxonomy at http://127.0.0.1:8000/admin/taxonomy/service/
3. Add at least 1 sector taxonomy at http://127.0.0.1:8000/admin/taxonomy/sector/
4. View the work listing page at http://127.0.0.1:8000/work/, the taxonomies you added above shouldn't be shown.
5. Similarly, when you view the blog listing at http://127.0.0.1:8000/blog/, you shouldn't see any of the taxonomies you added.
6. Now, you can edit any BlogPage by adding one of the new taxonomies you created earlier. When you view the blog listing at http://127.0.0.1:8000/blog/, you should see the new taxonomy displayed alongside the others.
7. Repeat step 6 for a WorkPage, and verify that when you view http://127.0.0.1:8000/work/, you see the taxonomy displayed.

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on Linux
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [x] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

**Review by ianmeigh — COMMENTED** [review]
**Question:** It may not be relevant to this PR, but what outcome would we prefer if someone used a saved link to an outdated taxonomy (that no longer exists)? Currently, this displays no results, but would it be preferable to show all blog/work pages?

**Review by ianmeigh — APPROVED** [review]
This change looks good to me, adding Nick as a reviewer.

File: tbx/work/models.py [hunk 1]
```diff
@@ -409,8 +409,18 @@ def get_context(self, request, *args, **kwargs):
         except EmptyPage:
             works = paginator.page(paginator.num_pages)
 
-        related_sectors = Sector.objects.all()
-        related_services = Service.objects.all()
+        # Only show Sectors and Services that have been used
```
Comment (nimasmi) [inline tbx/work/models.py:412]: **Question**: These related objects are being calculated based on `self.works`, after `works` has been further filtered in the local context. Is that the desired behaviour (i.e. do not consider the local filtering when deciding what labels to show)?

Ditto regarding blog posts.

**Review by nimasmi — APPROVED** [review]

File: tbx/work/models.py (hunk 1 above)
Comment (engineervix) [reply tbx/work/models.py:412]: Yes, we want to show all the labels regardless of the filtering that's been applied

**Pull request merged**